### PR TITLE
chore: added base project and icons

### PR DESCRIPTION
**Pull request by Shahroz16**
Part of https://github.com/customerio/issues/issues/9844

Complete each step to get your pull request merged in. [Learn more about the workflow this project uses](https://github.com/customerio/customerio-android/develop/docs/dev-notes/GIT-WORKFLOW.md).
- [ ] [Assign members of your team](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to review the pull request.
- [ ] Wait for pull request status checks to complete. If there are problems, fix them until you see that [all status checks are passing](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fsymfony.com%2Fdoc%2F4.3%2F_images%2Fdocs-pull-request-symfonycloud.png&f=1&nofb=1).
- [ ] Wait until the pull request has been reviewed *and approved* by a teammate
- [ ] After pull request is approved, and you determine it's ready **add the label "Ready to merge"** to the pull request and it will automatically be merged.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -18,6 +18,7 @@ buildscript {
         classpath Dependencies.gradleNexusPublishPlugin
         classpath Dependencies.dokka
         classpath Dependencies.kotlinGradlePlugin
+        classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.7.20'
```
Comment (mrehan27) [inline build.gradle:21]: Why do we need this plugin here? Isn't `Dependencies.kotlinGradlePlugin` the same? 🤔

File: settings.gradle [hunk 2]
```diff
@@ -5,3 +5,4 @@ include ':base'
 include ':messagingpush'
 include ':common-test'
 include ':messaginginapp'
+include ':kotlin_compose'
```
Comment (mrehan27) [inline settings.gradle:8]: Does this show up in Android project view if you open the SDK? Asking because it is placed in `samples` folder, and without specifying path I think it might not show sample code when opening the SDK.

I suggest updating as follows so we can group all sample apps inside sample directory in Android project view when opening the SDK in Android Studio

```suggestion
include ':samples:kotlin_compose'
```

File: kotlin_compose/src/main/ic_launcher-playstore.png
Comment (mrehan27) [inline kotlin_compose/src/main/ic_launcher-playstore.png:1]: Can you please double check why all these icons are not in `sample` folder?

**Review by mrehan27 — APPROVED** [review]

File: samples/kotlin_compose/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    id("org.jetbrains.kotlin.android")
+}
+
+android {
+    namespace = "io.customer.android.sample.kotlin_compose"
+    compileSdk = 33
+
+    defaultConfig {
+        applicationId = "io.customer.android.sample.kotlin_compose"
+        minSdk = 24
```
Comment (levibostian) [inline samples/kotlin_compose/build.gradle.kts:12]: Are we able to set the min to the min version of our SDK? I think this will improve the quality of our testing with these apps being able to test the min SDK version of our SDK is appropriate.

File: samples/kotlin_compose/src/main/res/values/themes.xml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+    <style name="Theme.CustomerioSDK" parent="android:Theme.Material.Light.NoActionBar" />
```
Comment (levibostian) [inline samples/kotlin_compose/src/main/res/values/themes.xml:4]: I thought that Jetpack Compose ignores `<style>` themes? I thought that themes were defined via Compose? 

So, is this file needed?

File: samples/kotlin_compose/build.gradle.kts (hunk 3 above)
Comment (Shahroz16) [reply samples/kotlin_compose/build.gradle.kts:12]: I think we can use it, but I think that would be more of a restriction rather than flexibility. Because we would want our SDK to be compiled against different SDK versions, if all our sample compiles against one then we lose the change to test it.

File: samples/kotlin_compose/.gitignore [hunk 5]
```diff
@@ -0,0 +1 @@
+/build
```
[the remark this answers is not in the document]
Comment (Shahroz16) [reply samples/kotlin_compose/.gitignore:1]: Good idea.

File: samples/kotlin_compose/src/main/res/values/themes.xml (hunk 4 above)
Comment (Shahroz16) [reply samples/kotlin_compose/src/main/res/values/themes.xml:4]: I plan to use [Splash screen](https://developer.android.com/develop/ui/views/launch/splash-screen), which requires `styles` for lower version APIs. 

Using this saves us from adding a custom new splash screen compasable and logic of how many X seconds it should stay etc.

File: samples/kotlin_compose/build.gradle.kts (hunk 3 above)
Comment (levibostian) [reply samples/kotlin_compose/build.gradle.kts:12]: I do think it's important that at least 1 of our sample apps uses the min SDK version. 

Maybe we make the Java sample app do that? We designate our Kotlin sample app to be more modern and use modern tools and SDK versions?

File: samples/kotlin_compose/src/main/res/values/themes.xml (hunk 4 above)
Comment (levibostian) [reply samples/kotlin_compose/src/main/res/values/themes.xml:4]: Do you believe that we need Splash screen? I want to avoid our sample apps having code that's irrelevant to QA or to customers using our SDK. Do you see a need for Splash screen and therefore this themes file?

File: samples/kotlin_compose/build.gradle.kts (hunk 3 above)
Comment (mrehan27) [reply samples/kotlin_compose/build.gradle.kts:12]: Java app already uses [21 for `minSdk`](https://github.com/customerio/customerio-android/blob/main/samples/java_layout/build.gradle#L19) which is same as our [`minSdk` in SDK](https://github.com/customerio/customerio-android/blob/main/buildSrc/src/main/kotlin/io.customer/android/Configurations.kt#L6).

Are you trying to say we should reference `Configurations.minSdk` from Java sample app as well?

Comment (levibostian) [reply samples/kotlin_compose/build.gradle.kts:12]: :facepalm: I apologize, I made a mistake that caused this confusion. 

For some reason, I was thinking the min sdk of our sdk was lower then 21. After checking it again, it is indeed 21 which matches the Java sample app as expected. 

Sorry for causing confusion here. We are good with both apps 👍

File: samples/kotlin_compose/src/main/res/values/themes.xml (hunk 4 above)
Comment (Shahroz16) [reply samples/kotlin_compose/src/main/res/values/themes.xml:4]: Splash is for the UX, these sample apps are not just going to be used by QA's but I think like before marketing team might use it to demo and test too. So having a black screen while the app loads seems like a bad experience to me and this is a very small tradeoff for it. 

And generally regarding 
> I want to avoid our sample apps having code that's irrelevant to QA or to customers

I think it varies, and I don't want all the sample apps to be same, because customers have different implementations themselves too. So its good if there are different approaches taken in different sample app and customers can relate to the one it finds closest to their implementation. 

We have had customers asking how they can use CIO using DI, so they won't have to replicate stuff etc, so the different variations don't hurt.
